Developers guide glossary cleanup (#583)
dfinity/docs: modules/developers-guide/pages/glossary.adoc

  A canister encapsulates all of the programming logic, public entry methods, the interface description for the provided message types, and state information for the dapp it describes.

[[g-principal]]
principal::
  A principal is an entity that can be authenticated by the {IC} blockchain. Principals that interact with the {IC} blockchain often do so via an identity. For example, the first time you use the {sdk-short-name}, the `+dfx+` command-line tool creates a `+default+` developer identity for you with a public/private key pair in a PEM file. This developer identity determines a **principal**. There is a textual representation of the principal which is often referred to as your **principal identifier**.
  Note that the developer identity can also be used to derive an **account identifier**—similar to a Bitcoin or Ethereum address—to hold ICP tokens on the identity's behalf in the ledger canister.

[[g-controller]]
controller::

>   A controller is a principal that has special rights to manage the canister it controls.

Reviewer: ```suggestion
  A controller is a principal that has special rights to manage the canister smart contract it controls.
```

[[g-ledger]]
ledger::
  The {IC} blockchain records all transactions involving ICP tokens in a specialized management canister smart contract, called the **ledger canister**.
  The ledger canister implements a smart contract that maintains **accounts** and **balances** and keeps a history of the *transactions* that affect accounts and balances. The transactions are recorded to track the following specific events:
  
  * **Mint ICP tokens** for accounts.
  * **Transfer ICP tokens** from one account to another.
  * **Burn ICP tokens** to eliminate them from existence.

[[g-wallet]]
wallet::
  A wallet is a specialized dapp (implemented as a canister smart contract) that allows to store and manage cycles. Cycles are used to pay for computation and resource consumption on the {IC} blockchain. Cycles can be obtained by conversion from the {IC} blockchain's native utility token ICP.

[[g-node]]
node::
  A physical or virtual machine that hosts all the hardware, replica software, and configuration settings required to participate in the {IC} blockchain.

[[g-replica]]
replica:: 

>   In the context of the {IC} blockchain, a replica refers to the {IC} protocol processes (for example, the `+replica+`, `+nodemanager+`) running on a node.

Reviewer: ```suggestion
  In the context of the {IC} blockchain, a replica refers to the {IC} protocol processes (for example, the `+replica+` process or the `+nodemanager+` processes) running on a node.
```

>   For the {sdk-short-name}, you use the `+dfx start+` and `+dfx stop+` commands to start and stop the `+replica+` process locally to provide a local network for development.

Reviewer: Is there now one or several processes that a make up a replica and which are started and stopped with dfx?

Author: Great point this is still somewhat confusing. How about this?
```
In the context of the {IC} blockchain, a replica refers to the {IC} protocol processes running on a node.
  For the {sdk-short-name}, you can use the `+dfx start+` and `+dfx stop+` commands to start and stop a `+replica+` process that provides a local network for development.
```

Author: Or maybe even something like this
```
In the context of the {IC} blockchain, a replica refers to the {IC} protocol processes running on a node. For the {sdk-short-name}, you can use the `+dfx start+` and `+dfx stop+` commands to start and stop a `+replica+` process that emulates a local network for development purposes.
```

Reviewer: I like the second one much better, but I'd use processes in plural in the second one as well (unless it's only one?)

Author: Good point. Done here 4fa6444.

[[g-wasm]]
WebAssembly::
  https://webassembly.org/[WebAssembly] (`+Wasm+`) is a low-level computer instruction format. 
  Because WebAssembly defines a portable, open-standard, binary format that abstracts cleanly over most modern computer hardware, it is broadly supported for programs that run on the internet.

>   Programs written in {proglang} are compiled to WebAssembly code for execution on replicas.

Reviewer: ```suggestion
  Dapps written in {proglang} are compiled to WebAssembly code for execution on the {IC} blockchain.
```
Maybe like this?

Author: Github won't let me apply this suggestion. Will manually apply.
